### PR TITLE
Feature/update game info

### LINKED ISSUE #115: PUT /games/{gameId} 修改遊戲資訊
參照 #32 
所有屬性都可以改 (`id` 除外)
但 `uniqueName` 必須要全域唯一，不可重複。

### PR DESCRIPTION
**Pull request by lohas1107**
## Why need this change? / Root cause: 
- As game maker, I want to update game information.
## Changes made:
- Game registration related
## Test Scope / Change impact:
- UpdateGameRegistrationUseCase
## Issue
- resolve #115

### CODE REVIEW

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/GameRegistrationControllerTest.kt [hunk 1]
```diff
@@ -202,4 +261,45 @@ class GameRegistrationControllerTest @Autowired constructor(
...
+
+    private fun givenGameHasRegistered(uniqueName: String, displayName: String): GameRegistration.Id {
+        val createGameRegistrationRequest = createGameRegistrationRequest(uniqueName, displayName)
+        return registerGameSuccessfully(createGameRegistrationRequest).id
+    }
+
+    private fun whenUpdateGameRegistration(gameId: String, updateGameRegistrationRequest: TestGameRegistrationRequest): ResultActions {
+        return mockMvc.perform(put("/games/$gameId")
+            .contentType(APPLICATION_JSON)
+            .content(updateGameRegistrationRequest.toJson()))
```
Comment (Wally5077) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/GameRegistrationControllerTest.kt:273]: **withJson(updateGameRegistrationRequest)**

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/GameRegistrationControllerTest.kt [hunk 2]
```diff
@@ -202,4 +261,45 @@ class GameRegistrationControllerTest @Autowired constructor(
...
+    }
+
+    private fun ResultActions.thenShouldReturnGameRegistrationNotFound() {
+        this.andExpect(status().isNotFound)
+            .andExpect(jsonPath("$.message").value("Resource (GameRegistration) not found (id = Id(value=not-exist-game-id))."))
+    }
+
+    private fun ResultActions.thenShouldReturnGameAlreadyExists() {
+        this.andExpect(status().isBadRequest)
```
Comment (Wally5077) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/GameRegistrationControllerTest.kt:282]: 全檔案都可以省略掉 **this.**

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/GameRegistrationControllerTest.kt [hunk 3]
```diff
@@ -202,4 +261,45 @@ class GameRegistrationControllerTest @Autowired constructor(
...
+            .andExpect(jsonPath("$.message").value("uno already exists"))
+    }
+
+    private fun ResultActions.thenUpdateGameRegistrationSuccessfully(request: TestGameRegistrationRequest) {
+        val result = this.andExpect(status().isOk)
```
Comment (Wally5077) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/it/controllers/GameRegistrationControllerTest.kt:287]: **result** -> **view**

**Review by Wally5077 — CHANGES_REQUESTED** [review]
**Review done.**

**Review by frankvicky — APPROVED** [review]

**Review by IamSaker — APPROVED** [review]

**Pull request merged**